pkg/server/server.go

```diff
@@ -686,10 +687,15 @@ func (s *ProxyServer) Connect(stream agent.AgentService_ConnectServer) error {
 				return
 			}
 
-			if s.warnOnChannelLimit && len(recvCh) >= xfrChannelSize {
+			select {
+			case recvCh <- in: // Send didn't block, carry on.
+			default: // Send blocked; record it and try again.
 				klog.V(2).InfoS("Receive channel on Connect is full", "agentID", agentID, "serverID", s.serverID)
```

Comment:
Note to reviewers: This log looks like it could get pretty spammy. I think there are a few options here:

1. It's not too bad, just leave it.
2. Keep the `--warn-on-channel-limit` flag, to limit the log line
3. Increase the verbosity level on this (e.g. `V(4)`)
4. Add some aggregation / rate limiting on the log line

I don't like the idea of having a flag just to control a spammy log line, so I'm leaning towards one of the other options.